dbms/src/Functions/FunctionsString.cpp

```diff
@@ -4573,7 +4576,14 @@ class FormatImpl : public IFunction
             using NumberFieldType = typename NumberType::FieldType;
             using NumberColVec = std::conditional_t<IsDecimal<NumberFieldType>, ColumnDecimal<NumberFieldType>, ColumnVector<NumberFieldType>>;
             const auto * number_raw = block.getByPosition(arguments[0]).column.get();
-            TiDBDecimalRoundInfo info{number_type, number_type};
+
+            auto prec = getDecimalPrecision(number_type, 0);
+            auto scale = getDecimalScale(number_type, 0);
+
+            /// output_scale is the second parameter of the 'format' function
+            auto output_scale = precision_base_col_size > 0 ? (*precision_base_col)[0].get<Int64>() : scale;
```

Comment:
What happens if `precision_base_col` is not a constant column, in this case the output_scale is differ from row to row.

Reply:
> What happens if `precision_base_col` is not a constant column, in this case the output_scale is differ from row to row.

When `precision_base_col` is not a constant column, we will always reset the `info.output_scale` with `max_num_decimals`  which can be shown in FunctionsString.cpp:4625.